routing/router.go

```diff
@@ -150,6 +150,14 @@ func newRouteTuple(amt lnwire.MilliSatoshi, dest []byte) routeTuple {
 	return r
 }
 
+// cntMutex is a struct that wraps a counter and a mutex, and is used
+// to keep track of the number of goroutines waiting for access to the
+// mutex, such that we can forget about it when the counter is zero.
+type cntMutex struct {
```

Comment:
The logic surrounding this struct could be further encapsulated using methods targeting the struct.  The type switch statements are already a bit bloated (when processing the anns), so would be nice to add as little as possible to them.